src/contracts/sgho/sGHO.sol

```diff
@@ -0,0 +1,380 @@
+   */
+  function initialize(
+    address gho_,
+    uint160 supplyCap_,
+    address executor_
+  ) public payable initializer {
+    if (gho_ == address(0) || executor_ == address(0)) revert ZeroAddressNotAllowed();
+
+    __ERC20_init('sGHO', 'sGHO');
+    __ERC4626_init(IERC20(gho_));
+    __ERC20Permit_init('sGHO');
+    __AccessControl_init();
+    __Pausable_init();
+    _grantRole(DEFAULT_ADMIN_ROLE, executor_);
+    _grantRole(PAUSE_GUARDIAN_ROLE, executor_);
+
+    sGHOStorage storage $ = _getsGHOStorage();
+    $.supplyCap = supplyCap_;
+    $.yieldIndex = RAY;
+    $.lastUpdate = uint64(block.timestamp);
+    $.ratePerSecond = 0; // Initial rate is 0, so ratePerSecond is 0 (no yield initially)
+    $.targetRate = 0;
+  }
+
+  /**
+   * @notice The receive function is implemented to reject direct Ether transfers to the contract.
+   * @dev sGHO does not handle ETH directly. All deposits must be made in the GHO token.
+   */
+  receive() external payable {
+    revert NoEthAllowed();
+  }
+
+  // --- ERC4626 Logic ---
+
+  /**
+   * @inheritdoc IERC4626
+   */
+  function maxWithdraw(address owner) public view override returns (uint256) {
+    if (paused()) {
+      return 0;
+    }
+
+    uint256 ghoBalance = IERC20(asset()).balanceOf(address(this));
+    uint256 maxWithdrawAssets = super.maxWithdraw(owner);
+    return maxWithdrawAssets < ghoBalance ? maxWithdrawAssets : ghoBalance;
+  }
+
+  /**
+   * @inheritdoc IERC4626
+   */
+  function maxRedeem(address owner) public view override returns (uint256) {
+    if (paused()) {
+      return 0;
+    }
+
+    uint256 ghoBalance = IERC20(asset()).balanceOf(address(this));
+    uint256 maxRedeemShares = super.maxRedeem(owner);
+    uint256 sharesForBalance = convertToShares(ghoBalance);
+    return maxRedeemShares < sharesForBalance ? maxRedeemShares : sharesForBalance;
+  }
+
+  /**
+   * @inheritdoc IERC4626
+   */
+  function maxDeposit(address) public view override returns (uint256) {
+    if (paused()) {
+      return 0;
+    }
+
+    sGHOStorage storage $ = _getsGHOStorage();
+    uint256 currentAssets = totalAssets();
+    return currentAssets >= $.supplyCap ? 0 : $.supplyCap - currentAssets;
+  }
+
+  /**
+   * @inheritdoc IERC4626
+   */
+  function maxMint(address receiver) public view override returns (uint256) {
+    return convertToShares(maxDeposit(receiver));
+  }
+
+  /**
+   * @inheritdoc IsGHO
+   */
+  function depositWithPermit(
+    uint256 assets,
+    address receiver,
+    uint256 deadline,
+    SignatureParams memory sig
+  ) external returns (uint256) {
+    // Use permit to approve the vault to spend the user's GHO tokens
+    try
+      IERC20Permit(asset()).permit(
+        _msgSender(),
+        address(this),
+        assets,
+        deadline,
+        sig.v,
+        sig.r,
+        sig.s
+      )
+    {} catch {}
+
+    return deposit(assets, receiver);
+  }
+
+  /**
+   * @inheritdoc IERC4626
+   */
```

Comment:
I d suggest to do this in general, but in thsi fn in particular I think it make sense to write its own docs as it's doing something different to 4626
```suggestion
  /// @inheritdoc IERC4626
```